### DISCUSSION TITLE
Azure automated tenant provisioning of Teams and SharePoint.

### DISCUSSION THREAD

**Discussion by paulwillen** [pnp/powershell#186] [Q&A] [answered]
Hi - 

My goal is to create an automated system which uses Azure Automation Account to provision Teams to an Office 365 tenant; update the connected SharePoint site with some default libraries; update the Team with default Team with additional channels and so on. 

You'd say that the command Invoke-PnPTenantTemplate is the best for this and so I'm building my soltion around that.
First I'm using the new PnP.PowerShell module. 
I registered an Azure Active Directory Application

`$aadApp = Register-PnPAzureADApp -ApplicationName "TeamsProvisioningAppSpecial" `
    -Tenant $tenant `
    -Store CurrentUser `
    -CommonName "TeamsProvisioningAppSpecial" `
    -CertificatePassword $certificatePassword `
    -ValidYears 5 `
    -OutPath C:temp -DeviceLogin`

and want to connect to the tenant using:

`Connect-PnPOnline $spoRoot -ClientId $clientId -Thumbprint $thumbprint -Tenant $tenant`

The problem is that after the command 
`Invoke-PnPTenantTemplate -Path .\tenant.xml `

I ALWAYS get the error: Invoke-PnPTenantTemplate: AADSTS70011: The provided request must include a 'scope' input parameter. The provided value for the input parameter 'scope' is not valid. The scope graph.microsoft.com/.default is not valid.

I have no idea where to look to solve this issue 
I found several blogposts and YouTube video's which are doing exactly what I'm doing (although with an older version of PnP.PowerShell), but no way it works for me and that's quite frustrating. 

Does anyone have the magic key to solve this puzzle for me. 

I'm using Paolo's XML-files described in this blogpost
https://www.collabmagazine.com/provisioning-teams-in-microsoft-teams-using-the-pnp-provisioning-engine/

**Comment by erwinvanhunen** [comment, marked as answer]
Which version of the cmdlets do you use? We have done fixes in PnP PowerShell that surfaced as the error you're describing.  Please install the latest nightly build, which should fix your issue:

```powershell
Update-Module -Name PnP.PowerShell -AllowPrerelease
```

**Reply by paulwillen** [reply]
Hi Erwin; 

Yes the nightly build 1.2.6 preview fixed the issue of the graph scope error. However I still cannot provision the team it seems. Going to look into that and if I cannot get it working I'll ask for help here..
